pkg/jobs/canary/canary_jobs.go

```diff
@@ -170,6 +181,99 @@ func (j CanaryJob) Run(ctx dutyjob.JobRuntime) error {
 	return nil
 }
 
+// formCheckRelationships forms check relationships with components and configs
+// based on the lookup expressions in the check spec.
+func formCheckRelationships(ctx context.Context, result *pkg.CheckResult) error {
+	check := result.Check
+	if result.Transformed {
+		check = result.ParentCheck // because the parent check has the relationship spec.
+	}
+
+	_check, ok := check.(RelatableCheck)
+	if !ok {
+		return nil
+	}
+	relationshipConfig := _check.GetRelationship()
+	if relationshipConfig == nil {
+		return nil
+	}
+
+	if result.Canary.GetCheckID(result.Check.GetName()) == "" {
+		logger.Tracef("no canary id found for check %s", result.Check.GetName())
+		return nil
+	}
+
+	checkID, err := uuid.Parse(result.Canary.GetCheckID(result.Check.GetName()))
+	if err != nil {
+		return fmt.Errorf("error parsing canary id(%s): %w", result.Canary.GetCheckID(result.Check.GetName()), err)
+	}
+
+	canaryID, err := uuid.Parse(result.Canary.GetPersistedID())
+	if err != nil {
+		return fmt.Errorf("error parsing canary id(%s): %w", result.Canary.GetPersistedID(), err)
+	}
+
+	for _, lookup := range relationshipConfig.Components {
+		if populated, err := lookup.Populate(result.Labels, map[string]any{"result": result}); err != nil {
+			logger.Errorf("error populating lookup config in check %s: %v", checkID, err)
+			continue
+		} else if !populated {
+			continue
+		}
+
+		logger.Infof("Finding all components with %s/%s/%s", lookup.Namespace.Value, lookup.Name.Value, lookup.Type.Value)
+		componentIDs, err := db.FindComponentIDsByNameNamespaceType(ctx, lookup.Namespace.Value, lookup.Name.Value, lookup.Type.Value)
+		if err != nil {
+			logger.Errorf("error finding components (check=%s) (lookup=%v): %v", checkID, lookup, err)
+			continue
+		}
+
+		for _, componentID := range componentIDs {
+			selectorID, err := utils.GenerateJSONMD5Hash(check)
```

Comment:
Not sure if this is appropriate to generate the relationship selector id

Reply:
Should the selector Id be a hash of the selector spec ?